hs_collection_resource/tests/test_collection.py

```diff
@@ -0,0 +1,414 @@
+        # again, the following two statements return wrong answers
+        # self.assertEqual(self.resCollection.raccess.public, False)
+        # self.assertEqual(self.resCollection.raccess.discoverable, False)
+
+    def test_custom_logic_in_hs_core_part_2(self):
+        # test custom logic for collection in hs_core
+        # this func tests that the public collection should downgrade to private if the last member res gets removed
+
+        # user 1 login
+        self.api_client.login(username='user1', password='mypassword1')
+
+        # no res file has been uploaded
+        self.assertFalse(self.resGen1.has_required_content_files())
+        # all required metadata terms have been provided when creating it
+        self.assertTrue(self.resGen1.metadata.has_all_required_elements())
+        self.assertFalse(self.resGen1.can_be_public_or_discoverable)
+
+        # check resGen1 current sharing status --> private
+        self.assertEqual(self.resGen1.raccess.public, False)
+        self.assertEqual(self.resGen1.raccess.discoverable, False)
+        self.assertEqual(BaseResource.objects.get(short_id=self.resGen1.short_id).raccess.public, False)
+        self.assertEqual(BaseResource.objects.get(short_id=self.resGen1.short_id).raccess.discoverable, False)
+
+        # add a res file to resGen1
+        files = [UploadedFile(file=self.res_file_obj, name=self.res_file_name)]
+        utils.resource_file_add_pre_process(resource=self.resGen1, files=files, user=self.user1,
+                                            extract_metadata=False)
+
+        utils.resource_file_add_process(resource=self.resGen1, files=files, user=self.user1,
+                                        extract_metadata=True)
+
+        # check resGen can_be_public_or_discoverable again --> True
+        self.assertTrue(self.resGen1.has_required_content_files())
+        self.assertTrue(self.resGen1.metadata.has_all_required_elements())
+        self.assertTrue(self.resGen1.can_be_public_or_discoverable)
+
+        # make resGen1 public
+        url_to_set_resource_flag_for_resGen1 = self.url_to_set_resource_flag.format(self.resGen1.short_id)
+        response = self.api_client.post(url_to_set_resource_flag_for_resGen1, {'t': 'make_public'}, HTTP_REFERER='http://foo/bar')
+
+        # check resGen1 new  sharing status --> public
+        self.assertEqual(BaseResource.objects.get(short_id=self.resGen1.short_id).raccess.public, True)
+        self.assertEqual(BaseResource.objects.get(short_id=self.resGen1.short_id).raccess.discoverable, True)
+        ## the following two statements return wrong sharing status. DONT KNOW WHY!!!
+        # self.assertEqual(self.resGen1.raccess.public, True)
+        # self.assertEqual(self.resGen1.raccess.discoverable, True)
+
+        # check resCollection current sharing status --> private
+        self.assertEqual(BaseResource.objects.get(short_id=self.resCollection.short_id).raccess.public, False)
```

Comment:
@pkdash for completeness i would like to keep this testing: we test the sharing status before and after our logic. i wont hurt anything...